src/main/java/ch/njol/skript/events/EvtMove.java

```diff
@@ -0,0 +1,104 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.coll.CollectionUtils;
+import io.papermc.paper.event.entity.EntityMoveEvent;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerMoveEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+public class EvtMove extends SkriptEvent {
```

Comment:
maybe this should be combined with EvtMoveOn?

syntax could be like:
`(0¦player|1¦%entitydatas%) (step|move|walk)[[ing] (on|over) %-itemtypes%]`

Reply:
due to that event having an executer and all sorts of other random stuff in it, I do not feel comfortable merging the two events. if you look at the class, its huge, and very very messy.

Reply:
fair enough ;D it can be done at a later date maybe